crowbar_framework/app/models/service_object.rb

```diff
@@ -960,33 +967,56 @@ def apply_role(role, inst, in_queue)
     new_elements = new_deployment["elements"]
     element_order = new_deployment["element_order"]
 
-    #
-    # Attempt to queue the proposal.  If delay is empty, then run it.
-    #
-    deps = proposal_dependencies(role)
-    delay, pre_cached_nodes = queue_proposal(inst, element_order, new_elements, deps)
-    unless delay.empty?
-      # force not processing the queue further
-      in_queue = true
-      # FIXME: this breaks the convention that we return a string; but really,
-      # we should return a hash everywhere, to avoid this...
-      return [202, delay]
+    # Build a list of old elements.
+    # elements_expanded on the old role is guaranteed to exists, as we already
+    # ran through apply_role with the old_role.  Cache is used for the case
+    # when pacemaker barclamp is deactivated.  elements_expanded gets updated
+    # by pacemaker barclamp.
+    old_elements = {}
+    old_deployment = old_role.override_attributes[@bc_name] unless old_role.nil?
+    unless old_deployment.nil?
+      old_elements = old_deployment["elements_expanded"]
+      if old_elements.nil?
+        old_elements = old_deployment["elements"]
+      end
     end
 
-    @logger.debug "delay empty - running proposal"
+    # When bootstrapping, we don't run chef, so there's no need for queuing
+    if bootstrap
```

Comment:
CI fails: `Failed to talk to service proposal commit: 405: {"error":"Failed to apply the proposal: uncaught exception (undefined local variable or method `bootstrap' for #\u003cCrowbarService:0x007f48801966b0\u003e)"}`